### PR TITLE
SolrCloud Configurations Proposal

### PR DESCRIPTION
**Pull request by cmharlow**
SolrCloud documentation, around practices & proposed procedures for interacting with SolrCloud resources.

See the series of PRs done around testing this option in funcake: https://github.com/tulibraries/funcake-solr/pulls?q=is%3Apr+is%3Aclosed

This is influenced / using work from the proposal PRs here:  https://github.com/tulibraries/tul_cob-az-solr/pull/7

This proposal is represented by PRs here:
- https://github.com/tulibraries/tul_cob-az-solr/pull/8
- https://github.com/tulibraries/tul_cob-web-solr/pull/7
- https://github.com/tulibraries/tul_cob-catalog-solr/pull/8

### CODE REVIEW

File: services/solrcloud.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+1. Local testing using SolrCloud docker-compose setup is performed to confirm the new Solr configurations do what you want.
+2. Cut the Solr Configset (on the appropriate Github repository; here, funcake-solr) release named X
+    1. CircleCI creates the following within SolrCloud:
+        * Configset named funcake-X
+        * Collection named funcake-X (largely ignored other than to create aliases)
+        * Aliases funcake-X-dev & funcake-X-prod (for tul_cob, this would be qa, stage, and prod aliases)
+3. Upon successful CircleCI deploy, Developer / Librarian: Goes to Airflow Production and performs the following:
+    1. Update variables for using new ConfigSets & Aliases, e.g.
+        * funcake-dev-solr-configset to funcake-X
+        * funcake-dev-solr-alias to funcake-X-dev
```
Comment (dkinzer) [inline services/solrcloud.md]: The alias is being auto created based on the configset name so this step is not required.

File: services/solrcloud.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+The following process SHOULD be followed for orchestrating Solr ConfigSet, Indexer Library, Solr Collections, Solr Aliases, & Solr Client updates:
+
+### Solr ConfigSet Release (does require changes in all downstream services)
+
+1. Local testing using SolrCloud docker-compose setup is performed to confirm the new Solr configurations do what you want.
+2. Cut the Solr Configset (on the appropriate Github repository; here, funcake-solr) release named X
+    1. CircleCI creates the following within SolrCloud:
+        * Configset named funcake-X
+        * Collection named funcake-X (largely ignored other than to create aliases)
```
Comment (bibliotechy) [inline services/solrcloud.md]:
```suggestion
        * Collection named funcake-X-init (largely ignored other than to create aliases)
```

File: services/solrcloud.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    1. Update variables for using new ConfigSets & Aliases, e.g.
+        * funcake-dev-solr-configset to funcake-X
+        * funcake-dev-solr-alias to funcake-X-dev
+        * If required / desired, update the funnel_cake_index traject configs branch
+    2. Run the funcake-dev dag to perform a full index using your new ConfigSet; Airflow performs the following:
+        * Harvest the OAI to s3
+        * Creates a new collection funcake-X-dev-TIMESTAMP using the new funcake-X configset
+        * Index the harvested data into funcake-X-dev-TIMESTAMP
+        * When harvest completes, updates alias funcake-X-dev to point to collection funcake-X-dev-TIMESTAMP
+4. While the funcake-dev dag Airflow Dag is running, the Developer or Librarian should create a new PR, with dev / qa as the base branch, on the client application (here, funnel_cake blacklight application) that does the following:
```
Comment (dkinzer) [inline services/solrcloud.md]:
```suggestion
4. While the funcake-dev dag Airflow Dag is running, the Developer or Librarian MUST create a new PR, with dev / qa as the base branch, on the client application (here, funnel_cake blacklight application) that does the following:
```

File: services/solrcloud.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            * stage configset & alias if appropriate
+            * funcake-prod-solr-configset to funcake-X
+            * funcake-prod-solr-alias to funcake-X-prod
+            * If required / desired, update the funnel_cake_index traject configs branch
+    2. Run the funcake-prod (or stage, then prod) dag to perform a full index using your new ConfigSet; Airflow performs the following:
+            * Harvest the OAI to s3
+            * Creates a new collection funcake-X-prod-TIMESTAMP using funcake-X configset
+            * Index the harvested data into funcake-X-prod-TIMESTAMP
+            * When harvest completes, updates alias funcake-X-prod to point to collection funcake-X-prod-TIMESTAMP
+    3. While the Airflow Dag is running, the Developer or Librarian should create a new PR on the client application (here, funnel_cake blacklight application) that merges the changes from dev to master (or qa to stage, then repeat step 6 for stage to production).
```
Comment (dkinzer) [inline services/solrcloud.md]:
```suggestion
    3. While the Airflow Dag is running, the Developer or Librarian MUST create a new PR on the client application (here, funnel_cake blacklight application) that merges the changes from dev to master (or qa to stage, then repeat step 6 for stage to production).
```

**Pull request merged**